File: 02-day-3--fibonacci-series/javascript/fibonacci.js
function fibonacci(num) {
  // type your code here
  let x = 0, y = 1
  let z
  if (num === 0){
    return 0
  }else{
    for (let i = 0; i<num; i++){
      z = x+y
      x = y
      y = z
    }
    return z
  }
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 0");
  console.log("=>", fibonacci(0));

  console.log("");

  console.log("Expecting: 1");
  console.log("=>", fibonacci(2));

  console.log("");

  console.log("Expecting: 55");
  console.log("=>", fibonacci(10));
}

module.exports = fibonacci;

// Please add your pseudocode to this file
// And a written explanation of your solution

File: 02-day-3--fibonacci-series/ruby/fibonacci.rb
def fibonacci(num)
  # type your code in here
  a, b = 0, 1
  if num == 0
    return 0
  end
  (2..num).each do |i|
    c = a + b
    a = b
    b = c
  end
  return b
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 0"
  puts "=>", fibonacci(0)

  puts

  puts "Expecting: 1"
  puts "=>", fibonacci(2)

  puts

  puts "Expecting: 55"
  puts "=>", fibonacci(10)

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# And a written explanation of your solution
